backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/KonanDefaultParameterInjector.kt

```diff
@@ -18,18 +18,27 @@ internal class KonanDefaultParameterInjector(private val konanContext: KonanBack
     override fun nullConst(startOffset: Int, endOffset: Int, type: IrType): IrExpression {
         val symbols = konanContext.ir.symbols
 
-        val nullConstOfEquivalentType = when (type.computePrimitiveBinaryTypeOrNull()) {
-            null -> IrConstImpl.constNull(startOffset, endOffset, context.irBuiltIns.nothingNType)
-            PrimitiveBinaryType.BOOLEAN -> IrConstImpl.boolean(startOffset, endOffset, type, false)
-            PrimitiveBinaryType.BYTE -> IrConstImpl.byte(startOffset, endOffset, type, 0)
-            PrimitiveBinaryType.SHORT -> IrConstImpl.short(startOffset, endOffset, type, 0)
-            PrimitiveBinaryType.INT -> IrConstImpl.int(startOffset, endOffset, type, 0)
-            PrimitiveBinaryType.LONG -> IrConstImpl.long(startOffset, endOffset, type, 0)
-            PrimitiveBinaryType.FLOAT -> IrConstImpl.float(startOffset, endOffset, type, 0.0F)
-            PrimitiveBinaryType.DOUBLE -> IrConstImpl.double(startOffset, endOffset, type, 0.0)
-            PrimitiveBinaryType.POINTER -> irCall(startOffset, endOffset, symbols.getNativeNullPtr.owner, emptyList())
-            PrimitiveBinaryType.VECTOR128 -> TODO()
-        }
+        val nullConstOfEquivalentType = type.unwrapToPrimitiveOrReference(
+                eachInlinedClass = { _, _ -> },
+                ifReference = { _ -> IrConstImpl.constNull(startOffset, endOffset, context.irBuiltIns.nothingNType) },
+                ifPrimitive = { primitiveType, nullable ->
+                    if (nullable)
+                        IrConstImpl.constNull(startOffset, endOffset, context.irBuiltIns.nothingNType)
```

Comment:
I'm not sure this is correct. Consider `type = Foo` where
```
inline class Foo(val value: NonNullNativePtr?)
```
Then `null` (of `Nothing?` type) doesn't conform to `Foo`.
Previous variant is probably wrong too, and I suspect this code is unreachable.

Reply:
(Discussed offline).